### ISSUE TITLE
Crash on opening video page

### ISSUE THREAD

**Issue by contrapunctus-1** [sschueller/peertube-android#45]
PeerTube v1.0.6 from F-Droid on a Nokia 6.1 running Android v9 (Pie).

Tapping on any video seems to crash the app.

Logs -
```
12-18 01:02:00.528 I/ActivityManager(1259): START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10300000 cmp=net.schueller.peertube/.activity.VideoListActivity (has extras)} from uid 10098
12-18 01:02:00.536 D/ActivityTrigger(1259): activityStartTrigger: Activity is Triggerred in full screen ApplicationInfo{b841f2e net.schueller.peertube}
12-18 01:02:00.536 E/ActivityTrigger(1259): activityStartTrigger: not whiteListednet.schueller.peertube/net.schueller.peertube.activity.VideoListActivity/106
12-18 01:02:00.539 D/ActivityTrigger(1259): activityResumeTrigger: The activity in ApplicationInfo{b841f2e net.schueller.peertube} is now in focus and seems to be in full-screen mode
12-18 01:02:00.539 E/ActivityTrigger(1259): activityResumeTrigger: not whiteListednet.schueller.peertube/net.schueller.peertube.activity.VideoListActivity/106
12-18 01:02:00.551 D/ActivityTrigger(1259): activityResumeTrigger: The activity in ApplicationInfo{b841f2e net.schueller.peertube} is now in focus and seems to be in full-screen mode
12-18 01:02:00.551 E/ActivityTrigger(1259): activityResumeTrigger: not whiteListednet.schueller.peertube/net.schueller.peertube.activity.VideoListActivity/106
12-18 01:02:00.555 D/ActivityManager(1259): Package [net.schueller.peertube] have process
12-18 01:02:00.558 I/PhoneWindow(1259): isAPPNeedChangeSBColor pkgName: net.schueller.peertube needKeep: false
12-18 01:02:00.559 I/PhoneWindow(1259): isAPPNeedChange pkgName: net.schueller.peertube needKeep: false
12-18 01:02:00.575 I/ActivityManager(1259): Start proc 21749:net.schueller.peertube/u0a22 for activity net.schueller.peertube/.activity.VideoListActivity
12-18 01:02:00.576 D/ActivityManager(1259): handleProcessStartedLocked mDoneFinishBooting && app.pid != 0 ---> notifyProcessStart, app.processName = net.schueller.peertube, app.info = ApplicationInfo{b841f2e net.schueller.peertube}, app.hostingType = activity, getHostingName() = ComponentInfo{net.schueller.peertube/net.schueller.peertube.activity.VideoListActivity}, getIntentFlag() = 0, getIsPackageRunning() = true, app = ProcessRecord{2e58c08 21749:net.schueller.peertube/u0a22}, getExtraData() = Bundle[{ACTIVITY_TASK_AFFINITY=net.schueller.peertube, ACTIVITY_INTENT=Intent { act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10300000 cmp=net.schueller.peertube/.activity.VideoListActivity (has extras) }, CALLER_PACKAGE=com.android.launcher3, ACTIVITY_INFO=ActivityInfo{44d1663 net.schueller.peertube.activity.VideoListActivity}}]
12-18 01:02:00.576 V/ActivityManager(1259): notifyProcessStart: net.schueller.peertube for: activity from: com.android.launcher3 packageName: net.schueller.peertube className: net.schueller.peertube.activity.VideoListActivity
12-18 01:02:00.622 I/[PowerSavingAppG3]BackgroundCleanService(3817): pid: 21749, calleeApp: net.schueller.peertube
12-18 01:02:00.623 I/[PowerSavingAppG3]ProcessMonitorService(3817): pid: 21749, calleeApp: net.schueller.peertube
12-18 01:02:00.626 I/[PowerSavingAppG3]BackgroundCleanService(3817): net.schueller.peertube add to mTopApList.
12-18 01:02:00.625 I/[PowerSavingAppG3]ProcessMonitorService(3817): net.schueller.peertube add to mTopApList.
12-18 01:02:00.712 I/PhoneWindow(21749): isAPPNeedChangeSBColor pkgName: net.schueller.peertube needKeep: false
12-18 01:02:00.714 I/PhoneWindow(21749): isAPPNeedChange pkgName: net.schueller.peertube needKeep: false
12-18 01:02:00.766 D/[PowerSavingAppG3]ProcessMonitorService(3817): topApps = [com.pluscubed.matloglibre, com.fastaccess.github.libre, net.schueller.peertube]
12-18 01:02:00.880 D/[PowerSavingAppG3]ProcessMonitorService(3817): calleePackageName = net.schueller.peertube,callerPacageName = com.android.launcher3,hostingType = activity,screenOn = true,cmp = net.schueller.peertube.activity.VideoListActivity,cat = {android.intent.category.LAUNCHER},action = android.intent.action.MAIN,isExceptionalActivitiy = false
12-18 01:02:00.884 D/[PowerSavingAppG3]ProcessMonitorService(3817): Ignore package net.schueller.peertube,component = net.schueller.peertube.activity.VideoListActivity
12-18 01:02:01.034 I/ActivityManager(1259): Displayed net.schueller.peertube/.activity.VideoListActivity: +479ms
12-18 01:02:01.199 W/SurfaceFlinger(763): Attempting to set client state on removed layer: Splash Screen net.schueller.peertube#0
12-18 01:02:01.200 W/SurfaceFlinger(763): Attempting to destroy on removed layer: Splash Screen net.schueller.peertube#0
12-18 01:02:03.398 I/ActivityManager(1259): START u0 {cmp=net.schueller.peertube/.activity.VideoPlayActivity (has extras)} from uid 10022
12-18 01:02:03.409 D/ActivityTrigger(1259): activityStartTrigger: Activity is Triggerred in full screen ApplicationInfo{b841f2e net.schueller.peertube}
12-18 01:02:03.409 E/ActivityTrigger(1259): activityStartTrigger: not whiteListednet.schueller.peertube/net.schueller.peertube.activity.VideoPlayActivity/106
12-18 01:02:03.411 D/ActivityTrigger(1259): activityResumeTrigger: The activity in ApplicationInfo{b841f2e net.schueller.peertube} is now in focus and seems to be in full-screen mode
12-18 01:02:03.411 E/ActivityTrigger(1259): activityResumeTrigger: not whiteListednet.schueller.peertube/net.schueller.peertube.activity.VideoPlayActivity/106
12-18 01:02:03.428 D/ActivityTrigger(1259): activityResumeTrigger: The activity in ApplicationInfo{b841f2e net.schueller.peertube} is now in focus and seems to be in full-screen mode
12-18 01:02:03.428 E/ActivityTrigger(1259): activityResumeTrigger: not whiteListednet.schueller.peertube/net.schueller.peertube.activity.VideoPlayActivity/106
12-18 01:02:03.454 I/PhoneWindow(21749): isAPPNeedChangeSBColor pkgName: net.schueller.peertube needKeep: false
12-18 01:02:03.456 I/PhoneWindow(21749): isAPPNeedChange pkgName: net.schueller.peertube needKeep: false
12-18 01:02:03.636 I/ActivityManager(1259): Displayed net.schueller.peertube/.activity.VideoPlayActivity: +195ms
12-18 01:02:03.905 E/AndroidRuntime(21749): Process: net.schueller.peertube, PID: 21749
12-18 01:02:03.905 E/AndroidRuntime(21749): java.lang.RuntimeException: Unable to start service net.schueller.peertube.service.VideoPlayerService@5cb92ba with Intent { cmp=net.schueller.peertube/.service.VideoPlayerService }: java.lang.SecurityException: Permission Denial: startForeground from pid=21749, uid=10022 requires android.permission.FOREGROUND_SERVICE
12-18 01:02:03.905 E/AndroidRuntime(21749): 	at net.schueller.peertube.service.VideoPlayerService$2.onNotificationStarted(VideoPlayerService.java:170)
12-18 01:02:03.905 E/AndroidRuntime(21749): 	at net.schueller.peertube.service.VideoPlayerService.playVideo(VideoPlayerService.java:183)
12-18 01:02:03.905 E/AndroidRuntime(21749): 	at net.schueller.peertube.service.VideoPlayerService.onStartCommand(VideoPlayerService.java:91)
12-18 01:02:03.908 W/ActivityManager(1259):   Force finishing activity net.schueller.peertube/.activity.VideoPlayActivity
12-18 01:02:03.917 W/ActivityManager(1259):   Force finishing activity net.schueller.peertube/.activity.VideoListActivity
12-18 01:02:03.926 I/ActivityManager(1259): Showing crash dialog for package net.schueller.peertube u0
12-18 01:02:04.366 I/StabilityMonitor(21832): [ParserUtility] key word : net.schueller.peertube
12-18 01:02:04.369 I/StabilityMonitor(21832): [APRReport] exceptionItem  tag:data_app_crash packageName:net.schueller.peertube versionCode:106 versionName:1.0.6
12-18 01:02:04.412 W/ActivityManager(1259): Activity pause timeout for ActivityRecord{fd88550 u0 net.schueller.peertube/.activity.VideoPlayActivity t20190 f}
12-18 01:02:04.426 E/STBIntentService(21832): send error message:data_app_crash;1545075123933(pkgName:net.schueller.peertube;version:1.0.6;versionCode:106) to BBS for ANR/FC/TOMB_STONE...
12-18 01:02:04.435 E/STBMonitorQueue(21832): restore data:/data/system/dropbox/data_app_crash@1545074834592.txt;106;1.0.6;net.schueller.peertube into regularQue queue...
12-18 01:02:04.435 E/STBMonitorQueue(21832): restore data:/data/system/dropbox/data_app_crash@1545074835895.txt;106;1.0.6;net.schueller.peertube into regularQue queue...
12-18 01:02:04.435 E/STBMonitorQueue(21832): restore data:/data/system/dropbox/data_app_crash@1545074859340.txt;106;1.0.6;net.schueller.peertube into regularQue queue...
12-18 01:02:04.435 E/STBMonitorQueue(21832): restore data:/data/system/dropbox/data_app_crash@1545075060024.txt;106;1.0.6;net.schueller.peertube into regularQue queue...
12-18 01:02:04.435 E/STBMonitorQueue(21832): restore data:/data/system/dropbox/data_app_crash@1545075061292.txt;106;1.0.6;net.schueller.peertube into regularQue queue...
12-18 01:02:04.436 E/STBMonitorQueue(21832): restore data:/data/system/dropbox/data_app_crash@1545075067014.txt;106;1.0.6;net.schueller.peertube into regularQue queue...
12-18 01:02:04.465 D/[PowerSavingAppG3]ProcessMonitorService(3817): topApps = [net.schueller.peertube, com.pluscubed.matloglibre, com.fastaccess.github.libre]
12-18 01:02:04.617 D/[PowerSavingAppG3]ProcessMonitorService(3817): topApps = [net.schueller.peertube, com.pluscubed.matloglibre, com.fastaccess.github.libre]
12-18 01:02:05.578 W/MediaCodec(21749): writeStatsLog:net.schueller.peertube
12-18 01:02:05.626 I/[PowerSavingAppG3]ProcessMonitorService(3817): original mTopApList = [com.fastaccess.github.libre, com.pluscubed.matloglibre, net.schueller.peertube]
12-18 01:02:05.627 I/[PowerSavingAppG3]ProcessMonitorService(3817): keep mTopApList = [com.fastaccess.github.libre, com.pluscubed.matloglibre, net.schueller.peertube]
12-18 01:02:05.663 W/ActivityManager(1259):   Force finishing activity net.schueller.peertube/.activity.VideoPlayActivity
12-18 01:02:05.664 W/ActivityManager(1259):   Force finishing activity net.schueller.peertube/.activity.VideoListActivity
12-18 01:02:05.676 W/SurfaceFlinger(763): Attempting to destroy on removed layer: a37ae68 net.schueller.peertube/net.schueller.peertube.activity.VideoPlayActivity#0
12-18 01:02:05.716 I/ActivityManager(1259): Killing 21749:net.schueller.peertube/u0a22 (adj 0): crash
12-18 01:02:05.717 W/ActivityManager(1259): Bringing down service while still waiting for start foreground: ServiceRecord{1c6f75a u0 net.schueller.peertube/.service.VideoPlayerService}
12-18 01:02:05.726 W/SurfaceFlinger(763): Attempting to destroy on removed layer: 4db0852 net.schueller.peertube/net.schueller.peertube.activity.VideoListActivity#0
12-18 01:02:10.778 D/[PowerSavingAppG3]ProcessMonitorService(3817): topApps = [com.pluscubed.matloglibre, net.schueller.peertube, com.fastaccess.github.libre]
12-18 01:02:14.103 I/[PowerSavingAppG3]ProcessMonitorService(3817): original mTopApList = [com.fastaccess.github.libre, net.schueller.peertube, com.pluscubed.matloglibre]
12-18 01:02:14.103 I/[PowerSavingAppG3]ProcessMonitorService(3817): keep mTopApList = [com.fastaccess.github.libre, net.schueller.peertube, com.pluscubed.matloglibre]
12-18 01:02:20.738 D/[PowerSavingAppG3]ProcessMonitorService(3817): topApps = [com.pluscubed.matloglibre, net.schueller.peertube, com.fastaccess.github.libre]
12-18 01:02:30.757 D/[PowerSavingAppG3]ProcessMonitorService(3817): topApps = [com.pluscubed.matloglibre, net.schueller.peertube, com.fastaccess.github.libre]
```

**Comment by sschueller** [comment]
Thanks, looks like I need the permission `android.permission.FOREGROUND_SERVICE` in Pie.

**Comment by sschueller** [comment]
Should be fixed in `v1.0.8` which should show up in f-droid within 24 hours.
